### PR TITLE
Remove support for per-instruction memory statistics

### PR DESCRIPTION
**Pull request by akiss77**
The new CBC interpreter does not support it anymore.

JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/jerry.h [hunk 1]
```diff
@@ -36,8 +36,7 @@ extern "C"
 #define JERRY_FLAG_SHOW_OPCODES           (1u << 0) /**< dump byte-code to stdout after parse */
 #define JERRY_FLAG_MEM_STATS              (1u << 1) /**< dump memory statistics */
 #define JERRY_FLAG_MEM_STATS_PER_OPCODE   (1u << 2) /**< dump per-instruction memory statistics during execution
-                                                     *   (in the mode full GC is performed after execution of each
-                                                     *   opcode handler) */
+                                                     *   FIXME: Remove. */
```
Comment (LaszloLango) [inline jerry-core/jerry.h]: Why only comment?

Comment (akiss77) [inline jerry-core/jerry.h]: I wasn't sure whether I should renumber all unrelated flags or leave an unused hole in the bits. But I'd go for the renumbering then.

Comment (LaszloLango) [inline jerry-core/jerry.h]: Could you also change the style of the flags to be an enum?

Comment (akiss77) [inline jerry-core/jerry.h]: Note that these are bit flags that can be combined (or'd). Do we use enums for such constructs? If we'd like to go for a change then bitfield structs seem a better fit to me. If.

Comment (LaszloLango) [inline jerry-core/jerry.h]: See `re_flags_t` and `ecma_property_flags_t`

Comment (zherczeg) [inline jerry-core/jerry.h]: I also prefer enum. see vm/vm.h
LGTM after that

**Comment by akiss77** [comment]
Updated according to the reviews.

**Comment by LaszloLango** [comment]
LGTM

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**